internal/p2p/peermanager.go

```diff
@@ -1175,9 +1299,47 @@ func (s *peerStore) Ranked() []*peerInfo {
 		s.ranked = append(s.ranked, peer)
 	}
 	sort.Slice(s.ranked, func(i, j int) bool {
-		// FIXME: If necessary, consider precomputing scores before sorting,
-		// to reduce the number of Score() calls.
-		return s.ranked[i].Score() > s.ranked[j].Score()
+		// sort inactive peers after active peers
+		if s.ranked[i].Inactive && !s.ranked[j].Inactive {
+			return false
+		} else if !s.ranked[i].Inactive && s.ranked[j].Inactive {
+			return true
+		}
+
+		iLastDialed, iLastDialSuccess := s.ranked[i].LastDialed()
+		jLastDialed, jLastDialSuccess := s.ranked[j].LastDialed()
+
+		// sort peers who our most recent dialing attempt was
+		// successful ahead of peers with recent dialing
+		// failures
+		switch {
+		case iLastDialSuccess && jLastDialSuccess:
+			// if both peers were (are?) successfully
+			// connected, convey their score, but give the
+			// one we dialed successfully most recently a bonus
+
+			iScore := s.ranked[i].Score()
+			jScore := s.ranked[j].Score()
+			if jLastDialed.Before(iLastDialed) {
+				jScore++
+			} else {
+				iScore++
+			}
+
+			return iScore > jScore
+		case iLastDialSuccess:
+			return true
+		case jLastDialSuccess:
+			return false
+		default:
+			// if both peers were not successful in their
+			// most recent dialing attempt, fall back to
+			// peer score.
+
+			// FIXME: If necessary, consider precomputing scores before sorting,
```

Comment:
I don't think it's too large a concern. I expect the loop in Score will typically have only one iteration anyway (or if not, a small handful).

But since we need these values on the above branch too, we could also hoist the `iScore` and `jScore` assignments out of the switch and reuse them in both places.

Reply:
oh yeah, this is just a legacy comment.